functionary/functionary/settings/debug.py

```diff
@@ -16,3 +16,8 @@
     "127.0.0.1",
     "10.0.2.2",
 ]
+
+try:
+    from .local import *
```

Comment:
I think you can solve the problem you were having by just importing the base stuff in local.py, and then putting this import immediately after the import from base here so:

```python
# local.py
from .base import *

INSTALLED_APPS += ["django_browser_reload"]
SOMESETTING = "HEY"
```

```python
# debug.py
import socket

from .base import *  # noqa
from .local import *  # noqa

# other stuff as usual
```

The fact that both local.py and debug.py import from base shouldn't cause a problem.  Also, I'd say just check in a local.py that has the base import and nothing else (maybe it has a comment explaining what it's for).  Then you don't need the try / except here and can instead just count on the file always being there.

And finally, add the local.py to the .gitignore so that people aren't accidentally checking in their personal settings.

Reply:
The .gitignore already had local_settings.py included so I moved it to that filename.

I left the local_settings import last so it can override all settings.